libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/map/MapFpsDelegate.java

```diff
@@ -0,0 +1,121 @@
+package com.mapbox.services.android.navigation.ui.v5.map;
+
+import android.content.Context;
+
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.services.android.navigation.ui.v5.camera.NavigationCamera;
+import com.mapbox.services.android.navigation.v5.navigation.MapboxNavigation;
+import com.mapbox.services.android.navigation.v5.navigation.NavigationConstants;
+import com.mapbox.services.android.navigation.v5.routeprogress.ProgressChangeListener;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteLegProgress;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
+
+class MapFpsDelegate {
+
+  private static final int VALID_DURATION_IN_SECONDS_UNTIL_NEXT_MANEUVER = 7;
+  private static final int VALID_DURATION_IN_SECONDS_SINCE_PREVIOUS_MANEUVER = 5;
+  private static final int DEVICE_MAX_FPS = 120;
```

Comment:
> This should be set to the maximum frame rate that the device can achieve reliably.

I don't feel like it's achievable to establish that based on a vast Android devices pool. I'd rather make this constants `Integer.MAX_VALUE` and let the hardware output frames as fast as it can. I'm sure there are, or will be in the future, devices that will be able to push more than 120 frames per second.

Reply:
Yeah I couldn't find a simple way to get the device FPS capability - setting to `Integer.MAX_VALUE` makes sense, as it will scale as Android screens get faster 😄 

Thank you both for this feedback 💯